integration/spark/src/test/java/io/openlineage/spark/agent/lifecycle/LogicalPlanSerializerTest.java

```diff
@@ -0,0 +1,344 @@
+import org.apache.spark.sql.catalyst.expressions.NamedExpression;
+import org.apache.spark.sql.catalyst.plans.logical.Aggregate;
+import org.apache.spark.sql.execution.datasources.CatalogFileIndex;
+import org.apache.spark.sql.execution.datasources.HadoopFsRelation;
+import org.apache.spark.sql.execution.datasources.InsertIntoDataSourceCommand;
+import org.apache.spark.sql.execution.datasources.InsertIntoHadoopFsRelationCommand;
+import org.apache.spark.sql.execution.datasources.LogicalRelation;
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions;
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCRelation;
+import org.apache.spark.sql.execution.datasources.text.TextFileFormat;
+import org.apache.spark.sql.internal.SQLConf;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StringType$;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.Test;
+import org.postgresql.Driver;
+import scala.Option;
+import scala.Tuple2;
+import scala.collection.Seq$;
+import scala.collection.immutable.HashMap;
+import scala.collection.immutable.Map$;
+
+class LogicalPlanSerializerTest {
+  private final TypeReference<Map<String, Object>> mapTypeReference =
+      new TypeReference<Map<String, Object>>() {};
+
+  @Test
+  public void testSerializeLogicalPlan() throws IOException {
+    SparkSession session = SparkSession.builder().master("local").getOrCreate();
+    String jdbcUrl = "jdbc:postgresql://postgreshost:5432/sparkdata";
+    String sparkTableName = "my_spark_table";
+    JDBCRelation relation =
+        new JDBCRelation(
+            new StructType(
+                new StructField[] {
+                  new StructField("name", StringType$.MODULE$, false, Metadata.empty())
+                }),
+            new Partition[] {},
+            new JDBCOptions(
+                jdbcUrl,
+                sparkTableName,
+                Map$.MODULE$
+                    .newBuilder()
+                    .$plus$eq(Tuple2.apply("driver", Driver.class.getName()))
+                    .result()),
+            session);
+    LogicalRelation logicalRelation =
+        new LogicalRelation(
+            relation,
+            Seq$.MODULE$
+                .<AttributeReference>newBuilder()
+                .$plus$eq(
+                    new AttributeReference(
+                        "name",
+                        StringType$.MODULE$,
+                        false,
+                        Metadata.empty(),
+                        ExprId.apply(1L),
+                        Seq$.MODULE$.<String>empty()))
+                .result(),
+            Option.empty(),
+            false);
+    Aggregate aggregate =
+        new Aggregate(
+            Seq$.MODULE$.<Expression>empty(),
+            Seq$.MODULE$.<NamedExpression>empty(),
+            logicalRelation);
+    ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
_Minor:_ Should we be create an `ObjectMapper` instance using [`OpenLineageClient.createMapper()`](https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/src/main/java/io/openlineage/spark/agent/client/OpenLineageClient.java#L155)